### ISSUE TITLE
AG WG Review: 2.1.3 Keyboard (No Exception) (Level AAA) WCAG2ICT Guidance

### ISSUE THREAD

**Issue by pday1** [w3c/wcag2ict#896]
## Summary
Review of proposed WCAG2ICT guidance for the Level AAA WCAG SC: 2.1.3 Keyboard (No Exception), as incorporated into the Editor's draft. 

## Maturity Level
Refining

## Action Needed
* Review content (See Content for Review section below for details.)
* Use the thumbs up and thumbs down icons to indicate agreement or disagreement with various comments. Do not add comments to simply support other comments. 
* Note any suggested edits for improvement and/or reasoning either in a comment on this issue or by creating a PR on the WCAG2ICT document.  
* Level A and AA SCs can be found in the [comments-by-guideline-and-success-criteria.md](https://github.com/w3c/wcag2ict/blob/main/comments-by-guideline-and-success-criterion.md) file.
* Level AAA SCs can be found in the [comments-on-level-AAA-success-criteria.md](https://github.com/w3c/wcag2ict/blob/main/comments-on-level-AAA-success-criteria.md)
----

## Background information 

Once finalized, this new WCAG2ICT Note will build on the previous version by also covering WCAG 2.2 level AAA success criteria (currently only level A and AA criteria are covered.  This is important as different jurisdictions are using different versions of WCAG for the accessibility of information and communication technology. 

## Content for Review

* Section [Applying SC 2.1.3 Keyboard (No Exception) to non-web documents and non-web software](https://w3c.github.io/wcag2ict/#applying-sc-2-1-3-keyboard-no-exception-to-non-web-documents).  

## Reference Material
In case you need them, links to various resources:

* [2.1.3 Keyboard (No Exception)](https://www.w3.org/TR/WCAG22/#keyboard-no-exception) SC text in WCAG 2.2
* [Intent section of Understanding 2.1.3 Keyboard (No Exception)](https://www.w3.org/WAI/WCAG22/Understanding/keyboard-no-exception#intent)

**Comment by maryjom** [comment]
I'm wondering if there should be an additional note to indicate that there are specific types of applications and use cases where mouse or pointer movement is deeply fundamental where it may be quite difficult to conform to this requirement, or may require alternate input and control methods to provide an alternate accessible experience. Some examples are:
- Digital art and design: where there may be freehand illustration and drawing or vector path manipulation.
- 3D Modeling and CAD: where 3-D sculpting tools must dynamically drag, pull, and smooth vertices across three dimensions simultaneously and CAD tools require exact point-to-point clicking on complex, multi-layered blueprints to snap lines and measure distances
- Medical and Scientific imaging: where imaging analysis tools require users to manually draw custom polygons around irregular boundaries or use smooth, omnidirectional panning and zooming to track microscopic anomalies
- Video editing and audio production: 
- Gaming and simulations: where applications such as Real-Time Strategy (RTS) and first-person aiming may require quickly creating bounding boxes over areas of the screen for taking action or require continuous 3-D rotational input where pointer speed and directionality that directly translate to camera movement.

**Comment by pday1** [comment]
Current guidance in editor's draft:

[Applying SC 2.1.3 Keyboard (No Exception) to non-web software](https://w3c.github.io/wcag2ict/#applying-sc-2-1-3-keyboard-no-exception-to-non-web-software)

> Where ICT is or includes non-web software that can be run on a software platform that provides a device-independent keyboard interface service, this applies directly as written, and as described in [Intent from Understanding Success Criterion 2.1.3](https://www.w3.org/WAI/WCAG22/Understanding/keyboard-no-exception.html#intent).
> 
> NOTE 1 (ADDED) (FOR NON-WEB SOFTWARE)
> Keyboard interface does not refer to a physical device but to the service of platform software (e.g. operating system, browser, etc.) that provides the software with keystrokes from any keyboard or keyboard substitute. When the non-web software supports such a device-independent service of the platform software, and the non-web software functionality is made fully operable through the service, then this success criterion would be satisfied.
> NOTE 2 (ADDED) (FOR NON-WEB SOFTWARE)
> A "device-independent keyboard interface service" refers to the platform service that provides keystrokes to any software running on the platform.
> NOTE 3 (ADDED) (FOR NON-WEB SOFTWARE)
> Inclusion of an on-screen keyboard can be done as well but does not satisfy this requirement since it does not allow for the use of keyboard alternatives whereas support of input from the device-independent keyboard interface service does.
> NOTE 4 (ADDED) (FOR NON-WEB SOFTWARE)
> This success criterion does not imply that non-web software always needs to directly support a keyboard or “keyboard interface” if one is not provided by the platform software. But if one is provided, the software needs to make all functionality available through it - unless the exception applies.
> NOTE 5 (ADDED) (FOR NON-WEB SOFTWARE)
> This success criterion also does not imply that non-web software always needs to provide its own virtual keyboard. But if it does, then the non-web software still needs to support keyboard input from any keyboard interface provided by the platform software.
> NOTE 6 (ADDED) (FOR NON-WEB SOFTWARE)
> For general guidance, see also the [comments on closed functionality](https://w3c.github.io/wcag2ict/#comments-on-closed-functionality). For commentary on the individual success criterion, see [Success criteria problematic for closed functionality, Keyboard (No Exception)](https://w3c.github.io/wcag2ict/#problematic-for-closed-keyboard-no-exception).

**Comment by pday1** [comment]
Proposal for discussion with TF:

NOTE (TO BE ADDED BEFORE CLOSED FUNCTIONALITY NOTE)
There are specific types of applications and use cases where mouse or pointer movement is deeply fundamental where it may be quite difficult to conform to this requirement, or may require alternate input and control methods to provide an alternate accessible experience. Examples include: 
- freehand illustration
- 3D modelling and sculpting
- gaming and simulation where pointer speed and directionality that directly translate to camera movement.
